specification/data-faas.md

```diff
@@ -0,0 +1,91 @@
+
+| Attribute name  | Notes  and examples  | Required? |
+|---|---|--|
+| `faas.trigger` | Type of the trigger on which the function is executed. <br > It SHOULD be one of the following strings: "datasource", "http", "pubsub", "timer", or "other". | Yes |
+| `faas.execution` | String containing the unique execution id of the function. E.g. `af9d5aa4-a685-4c5f-a22b-444f80b3cc28` | No |
+
+## Difference between execution and instance
+
+For performance reasons (e.g. [AWS lambda], or [Azure functions]), FaaS providers allocate an execution environment for a single instance of a function that is used to serve multiple requests.
+Developers exploit this fact to solve the **cold start** issue, caching expensive resource computations between different function executions. 
+Furthermore, FaaS providers encourage this behavior, e.g. [Google functions].
+Therefore, the span attribute `faas.execution` differs from the resource attribute `faas.instance` in the following:
+
+- `faas.execution` refers to the current request ID handled by the function;
+- `faas.instance` refers to the execution environment ID of the function.
+
+[AWS lambda]: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
+[Azure functions]: https://docs.microsoft.com/en-us/azure/azure-functions/manage-connections#static-clients
+[Google functions]: https://cloud.google.com/functions/docs/concepts/exec#function_scope_versus_global_scope
+
+# Function Type
+
+This section describes how to handle the span creation and additional attributes based on the value of the attribute `faas.trigger`.
+
+## Datasource
+
+A datasource function is triggered as a response to some data source operation such as a database or filesystem read/write.
+For `faas` spans with trigger `datasource`, it is recommended to set the following attributes. 
+
+| Attribute name  | Notes  and examples  | Required? |
+|---|---|--|
+| `faas.document.collection` | The name of the source on which the operation was perfomed. For example, in Cloud Storage or S3 corresponds to the bucket name, and in Cosmos DB to the database name. | Yes |
+| `faas.document.operation`  | Describes the type of the operation that was performed on the data.<br /> It SHOULD be one of the following strings: "insert", "edit", "delete". | Yes |
+| `faas.document.time`       | A string containing the time when the data was accessed in the [ISO 8601] format expressed in [UTC]. E.g. `"2020-01-23T13:47:06Z"` | Yes |
+| `faas.document.name`       | The document name/table subjected to the operation.<br /> For example, in Cloud Storage or S3 is the name of the file, and in Cosmos DB the table name.  | No |
+
+
+## HTTP
+The function responsibility is to provide an answer to an inbound HTTP request. The `faas` span MUST follow the recomandations described in the [HTTP Server semantic conventions](data-http.md#http-server-semantic-conventions).
```

Comment:
In case AWS-SDK invoke API falls into http trigger category, it cannot follow any HTTP semantic conventions